src/main/java/com/epam/aidial/core/service/LockService.java

```diff
@@ -35,13 +34,11 @@ public LockService(RedissonClient redis, @Nullable String prefix) {
     public Lock lock(String key) {
         String id = id(key);
         long owner = ThreadLocalRandom.current().nextLong();
-        long ttl = tryLock(id, owner);
-        long interval = WAIT_MIN;
 
-        while (ttl > 0) {
+        long interval = WAIT_MIN;
+        while (!tryLock(id, owner)) {
             LockSupport.parkNanos(interval);
             interval = Math.min(2 * interval, WAIT_MAX);
```

Comment:
Actually it should be used like Math.min( 2 * interval, Math.min(WAIT_MAX, ttl + 1))

Reply:
Changed.